js_v9/Block.js

```diff
@@ -78,21 +78,16 @@ function Block(fallingLane, color, iter, distFromHex, settled) {
 		if(attached === undefined)
 			attached = false;
 
-		if(this.angle > this.targetAngle) {
-			this.angularVelocity -= angularVelocityConst * MainHex.dt;
-		}
-		else if(this.angle < this.targetAngle) {
-			this.angularVelocity += angularVelocityConst * MainHex.dt;
-		}
-
-		if (Math.abs(this.angle - this.targetAngle + this.angularVelocity) <= Math.abs(this.angularVelocity)) { //do better soon
-			this.angle = this.targetAngle;
-			this.angularVelocity = 0;
-		}
-		else {
+		var sign = Math.sign(this.angle - this.targetAngle);
+		if (sign) {
+			this.angularVelocity -= sign * angularVelocityConst * MainHex.dt;
 			this.angle += this.angularVelocity;
+			if(sign !== Math.sign(this.angle + this.angularVelocity - this.targetAngle)) {
```

Comment:
I'm pretty sure this is to ensure that the hexagon doesn't overshoot the target angle.


> On Apr 17, 2015, at 1:48 PM, Isabell Cowan <notifications@github.com> wrote:
> 
> In js_v9/Block.js:
> 
> >  			this.angle += this.angularVelocity;
> > +			if(sign !== Math.sign(this.angle + this.angularVelocity - this.targetAngle)) {
> I'm honestly not sure why this. angle + this.angularVelocity needs to be computed again. It doesn't work without it, and it was essentially used in the original at line 88.
> 
> —
> Reply to this email directly or view it on GitHub.
>